src/components/UpgradePlanForm/UpgradePlanForm.js

```diff
@@ -4,45 +4,66 @@ import { Formik, Form, Field, ErrorMessage } from 'formik';
 import { InjectAppServices } from '../../services/pure-di';
 import { Loading } from '../../components/Loading/Loading';
 import { SubmitButton } from '../form-helpers/form-helpers';
+import useTimeout from '../../hooks/useTimeout';
 
 const fieldNames = {
   selectedPlanId: 'selectedPlanId',
   message: 'message',
 };
 /** @type { import('../../services/doppler-legacy-client').DopplerLegacyClient } */
-const UpgradePlanForm = ({ handleClose, isSubscriber, dependencies: { dopplerLegacyClient } }) => {
+const UpgradePlanForm = ({
+  handleClose,
+  isSubscriber,
+  dependencies: { dopplerLegacyClient, appSessionRef, planService },
+}) => {
   const [state, setState] = useState({
     availablePlans: null,
     formIsValid: false,
     isLoading: true,
   });
   const intl = useIntl();
+  const createTimeout = useTimeout();
 
   useEffect(() => {
     const fetchData = async () => {
       const availablePlans = await dopplerLegacyClient.getUpgradePlanData(isSubscriber);
+      const userPlan = appSessionRef.current.userData.user.plan;
+      const availablePlanMajors = planService.getAvailablePlans(userPlan, availablePlans);
```

Comment:
You should get the `availablePlans` from within the `planService`, we have them there already. 
```suggestion
      const availablePlanMajors = planService.getAvailablePlans(userPlan);
```

Reply:
I talked about it, and we decided that this change should be to continue